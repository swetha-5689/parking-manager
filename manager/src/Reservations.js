import React from "react";
import Table from "react-bootstrap/Table";
import Container from "react-bootstrap/Container";
//import Form from "react-bootstrap/Form";
import InputGroup from "react-bootstrap/InputGroup";
import FormControl from "react-bootstrap/FormControl";
import ReservationTableRow from "./ReservationRow"
//import { useAuth0 } from "./react-auth0-spa";
import axios from "axios";
class Reservations extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: []
    };
    this.changeData = this.changeData.bind(this);
  }
  componentDidMount() {
    axios.get('http://localhost:4000/api/res/')
        .then(response => {
            this.setState({data: response.data});
        })
        .catch(function (error) {
            console.log(error);
        })
  }
  changeData() {
    axios.get('http://localhost:4000/api/res/')
        .then(response => {
            this.setState({data: response.data});
        })
        .catch(function (error) {
            console.log(error);
        })
  }
  resTable() {
    return this.state.data.map((response, i) => {
      return <ReservationTableRow obj={response} key={i} data = {this.changeData}/>;
    });
  }
  render() {
    return (
      <div>
        <Container>
          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text id="inputGroup-sizing-default">
                Username
              </InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl
              placeholder="Search"
              aria-label="Default"
              aria-describedby="inputGroup-sizing-default"
            />
          </InputGroup>
        </Container>
        <Container className="mb-4">
          <Table striped bordered hover>
            <thead>
              <tr>
                <th>First Name</th>
                <th>Last Name</th>
                <th>Username</th>
                <th>Reservation Type</th>
                <th>Paid Status</th>
                <th>Stay Period</th>
                <th>Reservation End Time</th>
                <th>Confirmation Number</th>
                <th>Action</th>
              </tr>
            </thead>
            <tbody>
              {this.state.data === [] && <td>no data to show</td>}
            {this.state.data !== [] && this.resTable()}             
            </tbody>
          </Table>
        </Container>
      </div>
    );
  }
}
export default Reservations;
