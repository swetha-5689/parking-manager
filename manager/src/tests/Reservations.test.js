import React from 'react';
import renderer from 'react-test-renderer';
import Reservations from '../Reservations';


  const ReservationsList = {
    custFName: "Neha",
    custLName: "Nelson",
    type: "Guaranteed",
    car: "E82RTY",
    email: "neha@example.com",
    username: "n_nelson",
    resID: 129652,
    Paid: false,
    startTime: Date("March 18, 2020 9:00 AM EST"),
    endTime: Date("March 18, 2020 12:00 PM EST"),
    reserveTime: Date("March 18, 2020 12:00 PM EST")
  };


it('renders correctly when there are no items', () => {
  const tree = renderer.create(<Reservations />).toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders correctly when there is one item', () => {
    const tree = renderer.create(<Reservations data = {ReservationsList}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
