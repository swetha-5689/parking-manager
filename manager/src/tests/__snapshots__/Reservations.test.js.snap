// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="mb-3 input-group"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
          id="inputGroup-sizing-default"
        >
          Username
        </span>
      </div>
      <input
        aria-describedby="inputGroup-sizing-default"
        aria-label="Default"
        className="form-control"
        placeholder="Search"
      />
    </div>
  </div>
  <div
    className="mb-4 container"
  >
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            First Name
          </th>
          <th>
            Last Name
          </th>
          <th>
            Username
          </th>
          <th>
            Reservation Type
          </th>
          <th>
            Paid Status
          </th>
          <th>
            Stay Period
          </th>
          <th>
            Reservation End Time
          </th>
          <th>
            Confirmation Number
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`renders correctly when there is one item 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="mb-3 input-group"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
          id="inputGroup-sizing-default"
        >
          Username
        </span>
      </div>
      <input
        aria-describedby="inputGroup-sizing-default"
        aria-label="Default"
        className="form-control"
        placeholder="Search"
      />
    </div>
  </div>
  <div
    className="mb-4 container"
  >
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            First Name
          </th>
          <th>
            Last Name
          </th>
          <th>
            Username
          </th>
          <th>
            Reservation Type
          </th>
          <th>
            Paid Status
          </th>
          <th>
            Stay Period
          </th>
          <th>
            Reservation End Time
          </th>
          <th>
            Confirmation Number
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
